app/models/portfolio.rb

```diff
@@ -21,6 +21,20 @@ class Portfolio < ApplicationRecord
     end
   end
 
+  before_undiscard do
+    # only restore the portfolio_items that were discarded within 30 seconds of the portfolio being discarded
+    items = portfolio_items.with_discarded.discarded.select { |item| (item.discarded_at.to_i - discarded_at.to_i).abs < 30 }
+
+    if items.map(&:undiscard).any? { |result| result == false }
+      portfolio_items.with_discarded.discarded.each do |item|
```

Comment:
It appears line 29 `portfolio_items.with_discarded.discarded.each do |item|` doesn't check for the time stamp. There is a possibility that an error could get reported on an item not in the original list on line 26.

Reply:
Yes good catch. I rushed too much adding this. It would show false errors in this state, updated it to only select the items taht are still discarded.